sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -266,22 +261,27 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession with SQLHelper
       case _ =>
         // Create a test case to run this case.
         test(testCase.name) {
-          runTest(testCase)
+          runSqlTestCase(testCase, listTestCases, runQueries)
         }
     }
   }
 
-  /** Run a test case. */
-  protected def runTest(testCase: TestCase): Unit = {
-    def splitWithSemicolon(seq: Seq[String]) = {
-      seq.mkString("\n").split("(?<=[^\\\\]);")
-    }
-
-    def splitCommentsAndCodes(input: String) = input.split("\n").partition { line =>
-      val newLine = line.trim
-      newLine.startsWith("--") && !newLine.startsWith("--QUERY-DELIMITER")
-    }
+  /** A test case. */
+  protected trait TestCase {
```

Comment:
can we move it back to the original place? https://github.com/apache/spark/pull/40449/files#diff-212b35a074b1eeb518aff4c47130c7ab9cc23856ea6ea62f083af4452c637b9aL178

Reply:
Done.